src/screens/SignUp/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react'
+import NavigationService from '../../navigationService'
+import { ImageBackground, Text, View, Image, TextInput, TouchableHighlight, ActivityIndicator, KeyboardAvoidingView } from 'react-native'
+import { Formik } from 'formik'
+import AppText from '../../helpers/AppText'
+import styles from './styles'
+import * as firebase from 'firebase'
+
+const SignUp = () => {
+  const [authenticating, setAuthenticating] = useState(false)
+  const [error, setError] = useState(null)
+
+  const handleSubmit = (values) => {
+    setAuthenticating(true)
+    firebase.auth().createUserWithEmailAndPassword(values.email, values.password)
+      .then(user => {
+        setError(null)
+        setAuthenticating(false)
+        NavigationService.navigate('Home')
+      })
+      .catch(error => {
+        setError(error)
+        setAuthenticating(false)
```

Comment:
When you need to do something in both cases (`then/catch`), use `finally`. This also applies to `try/catch` blocks:
```javascript
promise.then(() => {
  doSomethingOnlyWhenEverythingIsOK()
}).catch((err) => {
  doSomethingWhenAnErrorIsThrown(err)
}).finally(() => {
  alwaysDoThisNoMatterIfAnErrorWasThrownOrNot()
})

// with try/catch
try {
  await promise
  doSomethingOnlyWhenEverythingIsOK()
} catch (err) {
  doSomethingWhenAnErrorIsThrown(err)
} finally {
  alwaysDoThisNoMatterIfAnErrorWasThrownOrNot()
}
```